boards/BMS/Src/App/states/App_PreChargeState.c

```diff
@@ -20,11 +31,76 @@ static void
     PreChargeStateRunOnTick100Hz(struct StateMachine *const state_machine)
 {
     App_AllStatesRunOnTick100Hz(state_machine);
+
+    struct BmsWorld *world = App_SharedStateMachine_GetWorld(state_machine);
+    struct BmsCanTxInterface *can_tx      = App_BmsWorld_GetCanTx(world);
+    struct PreCharge *        pre_charge  = App_BmsWorld_GetPreCharge(world);
+    struct Accumulator *      accumulator = App_BmsWorld_GetAccumulator(world);
+    struct Clock *            clock       = App_BmsWorld_GetClock(world);
+
+    const uint32_t pre_charge_duration =
+        App_SharedClock_GetCurrentTimeInMilliseconds(clock) -
+        App_SharedClock_GetPreviousTimeInMilliseconds(clock);
+    const float max_pack_voltage =
+        App_Accumulator_GetMaxPackVoltage(accumulator);
+    const float tractive_system_voltage =
+        App_PreCharge_GetTractiveSystemVoltage(pre_charge);
+    const struct State *next_state = App_GetPreChargeState();
+
+    if (pre_charge_duration <= App_PreCharge_GetMinDurationMs(pre_charge))
+    {
+        if (tractive_system_voltage > (max_pack_voltage * 0.93f))
```

Comment:
@kirons96 on top of the precharge timeout I think we should also monitor the voltage in case the precharge resistor somehow gets shorted while we precharge to 90%